src/temgymbasic/components.py

```diff
@@ -210,10 +327,37 @@ def gui_wrapper():
 
 class Source(Component):
     def __init__(
-        self, z: float, tilt_yx: Tuple[float, float] = (0., 0.), name: Optional[str] = None
+        self, z: float,
+        random: bool = False,
+        tilt_yx: Tuple[float, float] = (0., 0.),
+        name: Optional[str] = None,
+        wavelength: Optional[float] = 0.0,
```

Comment:
If they are none, then I have problem when they go into the Rays class lower down, because the wavelengths in rays need to be calculated somehow with the way I have written it: See line 329: wavelength=np.ones((r.shape[1],)) * self.wavelength

Reply:
I am not sure what the best way to deal with this is... I would like the input to rays be optional too of course, but I can't think how to handle both cases in a nice way, so this is why I set it to 0.0